### DISCUSSION TITLE
View transitions on form submission

### DISCUSSION THREAD

**Discussion by lilnasy** [withastro/roadmap#682] [Proposal]
### Body

Originially posted as an issue here: https://github.com/withastro/astro/issues/8381

When submitting a `<form>` or `<form action="get">`, the browser sends a request for the document at the form's action attribute, serializing the form's data as URLSearchParams, for example:

```html
<h3>Filter by Stars</h3>
<form>
    <button name="stars" value="1">1 Star</button>
    <button name="stars" value="2">2 Stars</button>
    <button name="stars" value="3">3 Stars</button>
</form>
```
Clicking on the "1 Star" button in this form should request the document at the URL /?stars=1. This is equivalent to using an anchor link to navigate to /?stars=1, but is more idiomatic. However, when submitting this form with the new view transitions single-page router, it performs a full page reload.

The page should retrieve the new page from the get request and replace the content with animations and while persisting transition:persist elements, just like it does when navigating via <a> tags.

**Comment by matthewp** [comment]
Would people expect this to be the *default* or something you should opt into? We have `data-astro-reload` as a way to opt-out at the moment, so it would be ok that this happens by default, if that's what people think is correct.

**Comment by brunocascio** [comment]
Have the same problem but with links instead. How did you solve it with links? I mean, how to change the query params without a full page reload?

**Comment by matthewp** [comment]
This is going to be released in 3.5 as an opt-in feature. So closing as this has been solved without needing a full RFC!
